bundle/jsky.app.ot/src/main/scala/jsky/app/ot/viewer/SearchPanel.scala

```diff
@@ -0,0 +1,120 @@
+package jsky.app.ot.viewer
+
+import edu.gemini.pot.sp.{ISPNode, ISPObservation}
+import edu.gemini.spModel.data.ISPDataObject
+import edu.gemini.spModel.rich.pot.sp._
+
+import javax.swing.JOptionPane
+import javax.swing.event.DocumentEvent
+import javax.swing.event.DocumentListener
+
+import scala.swing._
+import scala.swing.GridBagPanel._
+import scala.swing.Swing.EmptyBorder
+import scala.swing.event.FocusLost
+
+import scalaz._
+import Scalaz._
+
+/** Defines a label and text box to use for searching for program nodes.
+  */
+class SearchPanel(statusFilter: StatusFilter, tree: SPTree) extends GridBagPanel {
+  import SearchPanel._
+
+  private var viewer = Option.empty[SPViewer]
+
+  def searchString: Option[String] =
+    Option(textBox.text).filterNot(_.isEmpty)
```

Comment:
Thanks, I looked at this again and decided to move the `trim` down to where the matcher is created so I could make quoted text match case exactly while unquoted search text ignores case.